guard/guard.go

```diff
@@ -268,3 +275,70 @@ func (g *packageManagerGuard) clearStatus() {
 
 	g.interaction.ClearStatus()
 }
+
+func (g *packageManagerGuard) handleManifestInstallation(ctx context.Context, parsedCommand *packagemanager.ParsedCommand) error {
+	g.setStatus("Extracting packages from manifest files")
+
+	// Create extractor with appropriate ecosystem
+	var ecosystem packagev1.Ecosystem
+	switch g.packageManager.Name() {
```

Comment:
You should let the package manager define its own ecosystem to avoid mistakes. For example, introduce a new function called `Ecosystem()` in package manager interface and have it return this value